signer/client/signer_trust.go

```diff
@@ -180,16 +179,7 @@ func (trust *NotarySigner) ListAllKeys() map[string]string {
 // CheckHealth checks the health of one of the clients, since both clients run
 // from the same GRPC server.
 func (trust *NotarySigner) CheckHealth(timeout time.Duration) error {
-
-	// Do not bother starting checking at all if the connection is broken.
-	if trust.clientConn.State() != grpc.Idle &&
-		trust.clientConn.State() != grpc.Ready {
-		return fmt.Errorf("Not currently connected to trust server.")
-	}
-
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
-	// the kmClient should point at the same server as the sClient, so a single
-	// healthcheck should suffice
```

Comment:
@cyli been wondering about that since we removed it. Would it be possible to have the endpoint defined in both APIs but have our signer satisfy both with a single function? I don't know if there's a use case for putting key management and signing behind different services with different permissions. I could imagine somebody wanting to lock down key management more strongly than signing, especially if it's possible to cause keys to be deleted.